python/ray/train/_internal/state/schema.py

```diff
@@ -134,6 +142,10 @@ class TrainRunInfo(BaseModel):
         description="The UNIX timestamp of the end time of this Train run. "
         "If null, the Train run has not ended yet."
     )
+    run_error: Optional[TrainRunError] = Field(
```

Comment:
should we just re-use the status_detail field for this?

Reply:
I don't know if we need to store the rank in this field, maybe instead, we should just put worker-level errors in the worker models.

Reply:
The `status_details` is only a string. If we put the stack trace there, where should we store the failure rank?

Reply:
1. we can include the rank in the status detail string near at the top.
2. we can include each error per worker in the worker model.